tests/countries/test_finland.py

```diff
@@ -165,6 +166,126 @@ def test_all_saints_day(self):
             "1957-11-01",
         )
 
+    def _test_unofficial_holiday(self, name, month_and_day, year_of_adoption):
+        self.assertHolidayName(
+            name,
+            Finland(categories=UNOFFICIAL),
+            (f"{year}-{month_and_day}" for year in range(year_of_adoption, 2031)),
+        )
+        self.assertNoHolidayName(
+            name,
+            Finland(years=year_of_adoption - 1, categories=UNOFFICIAL),
+        )
+
+    def test_architecture_day(self):
+        name = "Arkkitehtuurin ja muotoilun päiviä"
+        self._test_unofficial_holiday(name, "02-03", 2012)
```

Comment:
Here and below you can just inline names if you use them once
```suggestion
        self._test_unofficial_holiday("Arkkitehtuurin ja muotoilun päiviä", "02-03", 2012)
```